src/core/components/hv-doc/index.tsx

```diff
@@ -39,12 +40,36 @@ export default class HvDocState extends PureComponent<
       styles: null,
       url: null,
     };
+    this.backBehaviorElementCache = [];
   }
 
   render(): React.ReactNode {
     return (
-      <Contexts.DocStateContext.Provider
+      <Contexts.DocContext.Provider
         value={{
+          backBehaviorElements: {
+            add: (elements: Element[]): void => {
+              if (elements?.length > 0) {
+                elements.forEach(e => {
+                  this.backBehaviorElementCache.push(e);
+                });
+              }
+            },
+            get: (): Element[] => this.backBehaviorElementCache,
+            remove: (elements: Element[]): void => {
+              if (
+                elements?.length > 0 &&
+                this.backBehaviorElementCache.length > 0
+              ) {
+                elements.forEach(e => {
+                  const ind = this.backBehaviorElementCache.indexOf(e);
+                  if (ind > -1) {
+                    this.backBehaviorElementCache.splice(ind, 1);
+                  }
+                });
+              }
```

Comment:
Given the types of `elements` and `this.backBehaviorElementCache`, I think the checks for `elements?.length` and `this.backBehaviorElementCache.length` are not needed, and it might be more "elegant" to use `Array.reduce` to perform the mutation of `this.backBehaviorElementCache` only once:

```suggestion
              this.backBehaviorElementCache = elements.reduce((acc, e) => {
                const i = acc.indexOf(e);
                return i > -1
                  ? [...acc].splice(i, 1)
                  : acc;
                }, this.backBehaviorElementCache);
```

Reply:
Almost worked. The return from splice() is the array which was sliced out. I'm returning the inverse of that. I tried using toSpliced() but it wasn't available in this case.